google/cloud/internal/timer_queue.h

```diff
@@ -62,36 +63,42 @@ namespace internal {
  * for (auto& t : svc) t.join();
  * @endcode
  */
-class TimerQueue {
+class TimerQueue : public std::enable_shared_from_this<TimerQueue> {
  public:
-  TimerQueue() = default;
+  static std::shared_ptr<TimerQueue> Create();
+
   TimerQueue(TimerQueue const&) = delete;
   TimerQueue& operator=(TimerQueue const&) = delete;
+  TimerQueue(TimerQueue&&) = delete;
+  TimerQueue& operator=(TimerQueue&&) = delete;
 
   // Adds a timer to the queue.
   future<StatusOr<std::chrono::system_clock::time_point>> Schedule(
       std::chrono::system_clock::time_point tp);
 
   // Signals all threads that have called Service to return. Does not modify
-  // remaining timers. If desired, CancelAll can be called.
+  // remaining timers.
```

Comment:
"Does not modify remaining timers" is misleading, given than this results in them being cancelled.

Reply:
A made a larger pass through these comments.  Could you take a look?

Reply:
Nice.